File: ruby/test/mit_id_test.rb
# frozen_string_literal: true

require_relative 'test_helper'

class MitIdTest < Minitest::Test
  def setup
    @mock = MockAdapter.new
    @client = Idkollen::ClientBuilder.new('cid', 'sec').http_client(@mock.connection).build
  end

  def test_auth_pending
    @mock.enqueue(200, { 'status' => 'PENDING', 'id' => 'm1', 'url' => 'https://mitid.login' })
    result = @client.mit_id.auth(Idkollen::MitId::AuthRequest.new)
    assert_kind_of Idkollen::MitId::Pending, result
    assert_equal 'https://mitid.login', result.url
  end

  def test_sign_completed_with_sign_result
    @mock.enqueue(200, {
      'status' => 'COMPLETED', 'id' => 'm1', 'ssn' => '0101010000',
      'name' => 'Hans', 'givenName' => 'Hans', 'surname' => 'H',
      'signResult' => { 'checksum' => 'abc123' },
    })
    result = @client.mit_id.sign_status('m1')
    assert_kind_of Idkollen::MitId::Completed, result
    assert_kind_of Idkollen::MitId::SignResult, result.sign_result
    assert_equal 'abc123', result.sign_result.checksum
  end
end

File: dart/test/mitid_test.dart
import 'package:idkollen_client/idkollen_client.dart';
import 'package:test/test.dart';

import 'mock_client.dart';

void main() {
  late MockClient mock;
  late IdkollenClient client;

  setUp(() {
    mock = MockClient();
    client = IdkollenClientBuilder('cid', 'sec').baseUrl('https://x.test').httpClient(mock).build();
  });

  test('auth pending', () async {
    mock.enqueue(200, {
      'status': 'PENDING',
      'id': 'm1',
      'url': 'https://mitid.example/start',
    });
    final result = await client.mitId.auth(const MitIdAuthRequest());
    expect(result, isA<MitIdPending>());
    expect((result as MitIdPending).id, 'm1');
  });

  test('sign completed with sign result', () async {
    mock.enqueue(200, {
      'status': 'COMPLETED',
      'id': 'm2',
      'ssn': '1234567890',
      'name': 'Test User',
      'givenName': 'Test',
      'surname': 'User',
      'signResult': {'checksum': 'abc123'},
    });
    final result = await client.mitId.signStatus('m2');
    expect(result, isA<MitIdCompleted>());
    expect((result as MitIdCompleted).signResult?.checksum, 'abc123');
  });
}
